mmdet/datasets/builder.py

```diff
@@ -35,6 +55,20 @@ def build_dataset(cfg, default_args=None):
             build_dataset(cfg['dataset'], default_args), cfg['times'])
     elif isinstance(cfg['ann_file'], (list, tuple)):
         dataset = _concat_dataset(cfg, default_args)
+    elif '*' in cfg['ann_file']:
```

Comment:
Maybe it'd be better to do globbing first and then process as usual?

Reply:
changed to glob, but * is required to avoid endless recursion of `build_dataset`

Reply:
I thought about something like this:
```python
else:
        cfg['ann_file'] = glob.glob(cfg['ann_file'], recursive=True)
        assert len(cfg['ann_file']) > 0
        if len(cfg['ann_file']) == 1:
            cfg['ann_file'] = cfg['ann_file'][0]
            dataset = build_from_cfg(cfg, DATASETS, default_args)
        else:
            dataset = _concat_dataset(cfg, default_args)
```
What do you think?

Reply:
fixed